tests/define_hooks_test_project.py

```diff
@@ -28,20 +28,20 @@ def raise_error(operation_name, job):
 
 
 @_HooksTestProject.operation
-@_HooksTestProject.add_hook.on_start(set_job_doc(HOOK_KEYS[0]))
-@_HooksTestProject.add_hook.on_finish(set_job_doc(HOOK_KEYS[1]))
-@_HooksTestProject.add_hook.on_success(set_job_doc(HOOK_KEYS[2]))
-@_HooksTestProject.add_hook.on_fail(set_job_doc_with_error())
+@_HooksTestProject.operation_hook.on_start(set_job_doc(HOOK_KEYS[0]))
+@_HooksTestProject.operation_hook.on_finish(set_job_doc(HOOK_KEYS[1]))
+@_HooksTestProject.operation_hook.on_success(set_job_doc(HOOK_KEYS[2]))
+@_HooksTestProject.operation_hook.on_fail(set_job_doc_with_error())
```

Comment:
Looking at this, I am reminded of our discussion to require an ordering to our decorators, such as a function must be labelled an operation before applying other decorators providing specifics about those operators. Should we go ahead and make these decorators follow that decision? What are your thoughts @bdice? I am inclined to do this since we seem to be moving that way and then we don't have to deprecate the change for these methods later if we do require the ordering.

Reply:
I think this falls outside of the scope of this PR.

Reply:
I agree that is out of scope. The PR that would introduce a required order #538 has been stalled because of the complexity of enforcing decorator ordering.

Reply:
I would disagree #538 is stalled due to complexity, but we can punt on this until later.